### ISSUE TITLE
New Proto Files

### ISSUE THREAD

**Issue by barnjamin** [tsuna/gohbase#51]
I'd like to add Admin.proto to get access to the CompactRegion function but that requires several other files.  

It looks like  HBase is preparing for some changes internally.  There is no longer a `Master.proto` file in `hbase-protocol`, and there is now a `hbase-protocol-shaded` directory with overlapping proto files.

Given the changes and possible conflicts, how should I handle adding or updating proto files?

**Comment by barnjamin** [comment]
For now I've just checked out hbase `branch-1.2` and used those proto files but I get UnsupportedOperationException anytime i try to call one of the rpc functions in Admin.proto.  Any advice? https://github.com/apache/hbase/blob/branch-1.2/hbase-protocol/src/main/protobuf/Admin.proto#L255

**Comment by Timoha** [comment]
I'm not sure you can use `AdminService` functions directly from gohbase. Instead you should implement a new `hrpc.Call` using the `*Request`, `*Response` messages you want to use the same way you did for ClusterStatus. The `UnsupportedOperationException` might be returned because the function name you return in `Name()` might be unknown to master. Which method are you trying to implement?

**Comment by Timoha** [comment]
I'll update the proto files later today to use branch-1.3, unless you want to send a PR with that

**Comment by barnjamin** [comment]
Trying to implement `CompactRegion`

```
2017/06/21 12:54:39 Failed to compact region (entity,1e01addab1196173d135f74fc1f08cdf69e1efca,1479510494211.b8e38f9469037daff435b85f82dcfc09.): HBase Java exception org.apache.hadoop.hbase.DoNotRetryIOException:
/0.0.0.0:60000 is unable to read call parameter from client 10.10.7.147; java.lang.UnsupportedOperationException: CompactRegion
```

The value I'm sending for Name is the same as appears here: https://github.com/apache/hbase/blob/branch-1.3/hbase-protocol/src/main/protobuf/Admin.proto#L284

I can keep trying to read through the HBase source for why AdminService is different.

I will also submit a pr for 1.3 proto files.

**Comment by barnjamin** [comment]
https://github.com/apache/hbase/blob/branch-1.3/hbase-client/src/main/java/org/apache/hadoop/hbase/client/HBaseAdmin.java#L182 This may be why I'm unable to call AdminService methods

**Comment by Timoha** [comment]
Nah, I don't think so. They just say that you can't instantiate HBaseAdmin directly, and need to do conn.getAdmin() instead. Could you send a PR to check how you are doing a request?

**Comment by barnjamin** [comment]
I closed the PR (#55) already but the tools/localize.go is approximately what I'm trying to run when i get an UnsupportedOpException

**Comment by Timoha** [comment]
I'm going to close this issue as we already merged the new proto files. Feel free to open a new more relevant one.
